src/pe-case-and-asset-map/widget/pe-case-and-asset-map.u-client-script.js

```diff
@@ -0,0 +1,285 @@
+function mapPageCtrl($scope, $rootScope, $window,$location, $anchorScroll, $timeout, leafletData, $q) {
+	/* widget controller */
+	var c = this;
+	$scope.firstFlip = true;
+	$location.hash('ff');
+	
+	$rootScope.$broadcast('refreshFooterUrl');
+	
+	c.loading = true;
```

Comment:
Consider moving to $onInit method for initialization code & organization. See the style guide. https://github.com/platform-experience/serviceportal-best-practice#oninit